src/discretization/fist.jl

```diff
@@ -142,3 +142,15 @@ function isearccw(𝒫::Chain{Dim,T}, i) where {Dim,T}
 
   isconvex && !intersects && incones
 end
+
+function update_adjacent_ear!(𝒬, 𝒫, i, n)
+  if !isear(𝒫, i)
+    ind = findfirst(==(i), 𝒬)
+    !isnothing(ind) && deleteat!(𝒬, ind)
+  else
+    if mod1(i,n) ∉ 𝒬
+      ind = something(findfirst(>(mod1(i,n)), 𝒬), 1)
+      insert!(𝒬, ind, mod1(i,n))
+    end
+  end
```

Comment:
Perhaps we can simplify this code considerably. The logic seems very simple assuming we have a single function call to update both years:

```julia
function update_adjacent_ears!(Q, P, i)
  # check if adjacent ears are valid ears
  for j in [i-1, i+1]
    if valid(j)
      # add to Q
    else
      # remove from Q
    end
  end
end
```
I will try to refactor it locally.